packages/pnpmfile/src/requirePnpmfile.ts

```diff
@@ -50,8 +50,14 @@ export default (pnpmFilePath: string, prefix: string) => {
         pkg.peerDependencies = pkg.peerDependencies || {}
         const newPkg = readPackage(pkg, ...args)
         if (!newPkg) {
-          throw new BadReadPackageHookError(pnpmFilePath)
+          throw new BadReadPackageHookError(pnpmFilePath, 'readPackage hook did not return a package manifest object.')
         }
+        const dependencies = ['dependencies', 'optionalDependencies', 'peerDependencies']
+        dependencies.forEach(dep => {
```

Comment:
this function is called on each resolution, so on big projects it may be called a thousands of times. Might make sense to optimize it and use a `for of` loop instead